gammapy/makers/utils.py

```diff
@@ -472,26 +474,51 @@ def make_counts_rad_max(geom, rad_max, events):
     counts : `~gammapy.maps.RegionNDMap`
         Counts vs estimated energy extracted from the ON region.
     """
-    rad_max = get_rad_max_vs_energy(rad_max, events.pointing_radec, geom)
+    selected_events = apply_rad_max(events, rad_max, geom.region.center)
 
-    counts_list = []
+    counts_geom = RegionGeom(geom.region, axes=[geom.axes['energy']])
+    counts = Map.from_geom(counts_geom)
+    counts.fill_events(selected_events)
 
-    # create and fill a map per each energy bin, fetch the counts
-    for i, rad in enumerate(rad_max):
-        on_region = CircleSkyRegion(center=geom.center_skydir, radius=rad)
-        energy_range = geom.axes["energy"].slice(i)
-        on_region_geom = RegionGeom(on_region, axes=[energy_range])
-        counts = Map.from_geom(on_region_geom)
-        counts.fill_events(events)
-        counts_list.append(counts.data[0])
+    return counts
 
-    counts = RegionNDMap.from_geom(geom, data=np.asarray(counts_list))
 
-    return counts
+def apply_rad_max(events, rad_max, position):
+    '''Apply the RAD_MAX cut to the event list for given region'''
+    offset = position.separation(events.pointing_radec)
+    separation = position.separation(events.radec)
+
+    if rad_max.data.shape == (1, 1):
+        rad_max_for_events = rad_max.quantity[0, 0]
+    else:
+        rad_max_for_events = rad_max.evaluate(energy=events.energy, offset=offset)
+
+    selected = separation <= rad_max_for_events
+    return events.select_row_subset(selected)
+
+
+def are_regions_overlapping_rad_max(regions, rad_max, offset):
```

Comment:
OK, this specific implementation might remain. The region based alternative would be to create a list of `CircleSkyRegions` and then use again `np.any([regions_overlap(reg_1, reg_2, wcs=wcs) for reg_1, reg_2 in combinations(regions, 2)])`

However `regions_overlap` could implement the short cut for `CircleSkyRegion`, based on the separation. But of course the purely region based method will be slower...not sure it matters...

Reply:
On the other hand: having some functionality to create a list of circular regions from a `RadMax2D` object might be useful e.g. for plotting as well..